vignettes/introduction.Rmd

```diff
@@ -0,0 +1,335 @@
+```{r prepare, include = FALSE}
+# Restore the current instance at the end of the vignette
+current_instance <- laminr::get_current_lamin_instance()
+withr::defer(laminr::lamin_connect(current_instance))
+
+# Restore the current user at the end of the vignette
+current_user <- laminr::get_current_lamin_user()
+withr::defer(laminr::lamin_login(current_user))
+
+laminr::lamin_logout() # Use an anonymous user for this vignette
+```
+
+```{r setup}
+library(laminr)
+```
+
+# Walkthrough
+
+See https://docs.lamin.ai/guide#walkthrough.
+
+```{r walkthrough}
+lamin_init_temp(name = "lamin-intro", modules = c("bionty"))
+```
+
+**NOTE:** We have used a `lamin_init_temp()` to create a temporary instance for this vignette but in most cases you should use regular `lamin_init()`
+
+```{r disconnect, include = FALSE}
+# Disconnect from this instance at the end of the vignette
+withr::defer(laminr::lamin_disconnect())
+```
+
+## Transforms
+
+See https://docs.lamin.ai/guide#transforms.
+
+```{r transforms}
+ln <- import_lamindb()
+ln$track()
+
+ln$Transform$df()
+
+ln$Run$df()
+```
+
+## Artifacts
+
+See https://docs.lamin.ai/guide#artifacts.
+
+```{r artifacts}
+df <- ln$core$datasets$small_dataset1(otype = "DataFrame", with_typo = TRUE)
+df
+
+artifact <- ln$Artifact$from_df(df, key = "my_datasets/rnaseq1.parquet")$save()
+artifact$describe()
+
+artifact$cache()
+
+dataset <- artifact$open()
+as.data.frame(dataset)
+
+artifact$load()
+
+artifact$view_lineage()
+
+df_typo <- df
+levels(df$perturbation) <- c("DMSO", "IFNG")
+df["sample2", "perturbation"] <- "IFNG"
+artifact <- ln$Artifact$from_df(df, key = "my_datasets/rnaseq1.parquet")$save()
+artifact$versions$df()
+```
+
+## Labels
+
+See https://docs.lamin.ai/guide#labels.
+
+```{r labels}
+bt <- import_bionty()
+
+experiment_type <- ln$ULabel(name = "Experiment", is_type = TRUE)$save()
+candidate_marker_experiment <- ln$ULabel(
+  name = "Candidate marker experiment", type = experiment_type
+)$save()
+
+artifact$ulabels$add(candidate_marker_experiment)
+
+cell_type <- bt$CellType$from_source(name = "effector T cell")$save()
+artifact$cell_types$add(cell_type)
+
+artifact$describe()
+```
+
+## Registries
+
+See https://docs.lamin.ai/guide#registries.
+
+```{r registries}
+ln$ULabel$df()
+
+ln$Artifact
+```
+
+## Query & search
+
+See https://docs.lamin.ai/guide#query-search.
+
+```{r query-search}
+transform <- ln$Transform$get(key = "introduction.Rmd")
+
+ln$Artifact$filter(key__startswith = "my_datasets/")$df()
+
+artifacts <- ln$Artifact$filter(transform = transform)$all()
+
+artifacts <- ln$Artifact$filter(
+  transform__description__icontains = "intro", ulabels = candidate_marker_experiment
+)$all()
+
+ln$Transform$search("intro")$df()
+ulabels <- ln$ULabel$lookup()
+cell_types <- bt$CellType$lookup()
+```
+
+## Features
+
+See https://docs.lamin.ai/guide#features.
+
+```{r features}
+ln$Feature(name = "temperature", dtype = "float")$save()
+
+ln$Feature(name = "experiment", dtype = ln$ULabel)$save()
```

Comment:
Here re-creating a new "experiment" feature errors with different dtype, due to the previous line `experiment_type <- ln$ULabel(name = "Experiment", is_type = TRUE)$save()` already defined the feature with the same name.

It's not a bug, just we might need to refactor the tutorial a bit.